doc/DeepSpeech.md

```diff
@@ -0,0 +1,70 @@
+
```

Comment:
General comment on this approach:

> How will documentation in the `doc` directory stay in sync with the code?

Generally,  if the code and the documentation is in the same place, this process is "easy". However, here the code and documentation are in separate files. This doesn't seem to lend itself to keeping the documentation and code in sync.

Why was something like [Sphinx](http://www.sphinx-doc.org/en/stable/index.html) not used?